src/solver/ecg.jl

```diff
@@ -59,6 +40,147 @@ function Plonsey1964ECGGaussCache(problem::SplitProblem{<:TransientHeatProblem},
+    electrode_pairs::Vector{Tuple{Int, Int}}) where T
+
+    sdim = Ferrite.getdim(mesh)
+    ets = elementtypes(mesh)
+    @assert length(ets) == 1 "Multiple elements not supported yet." 
+
+    nleadfields = length(electrode_pairs)
+
+    ip = getinterpolation(ip_collection, getcells(mesh,1))
+    qr = getquadraturerule(qr_collection, getcells(mesh,1))
+    cv = CellValues(qr, ip);
+
+    dh = DofHandler(mesh)
+    Ferrite.add!(dh, :Z, ip)
+    close!(dh);
+    Z = Matrix{T}(undef, nleadfields, ndofs(dh))
+    κ∇φₘ = zeros(Vec{sdim, T}, getnquadpoints(qr), getncells(dh.grid))
+    lead_field = Geselowitz1989ECGLeadCache(Z, κ∇φₘ, dh, cv, zero_vertex, electrodes, electrode_pairs)
+    _compute_lead_field(lead_field, _κ)
+    return lead_field
+end
+
+function _compute_lead_field(lead_field::Geselowitz1989ECGLeadCache, _κ::SpectralTensorCoefficient)
+    
+    @unpack Z, dh, cv, zero_vertex, electrodes, electrode_pairs = lead_field
+
+    K = create_sparsity_pattern(dh)
+    f = zeros(ndofs(dh))
+
+    integrator = BilinearDiffusionIntegrator(_κ)
+    element_cache = BilinearDiffusionElementCache(integrator, cv)
+    op = AssembledBilinearOperator(K, element_cache, dh)
+    update_operator!(op, 0.)
+
+    ch = ConstraintHandler(dh)
+    Ferrite.add!(ch, Dirichlet(:Z, Set([zero_vertex]), (x, t) -> 0.0))
+    close!(ch);
+    apply!(op.A, f, ch)
+
+    for (i, electrode_pair) in enumerate(electrode_pairs)
+        _add_electrodes(f, dh, "lead_field_$(i)", electrodes[electrode_pair[1]], electrodes[electrode_pair[2]])
+        ml = ruge_stuben(op.A)
+        p = aspreconditioner(ml)
+        Z_lead = @view Z[i, :]
+        IterativeSolvers.cg!(Z_lead, op.A, f, Pl=p)
+    end    
+end
+
+function _add_electrodes(f::Vector{T}, dh::DofHandler, set_name::String, positive_electrode::Vec{3,T}, negative_electrode::Vec{3,T}) where T<:Number
+    mesh = dh.grid
+    # Todo: clean this
+    haskey(mesh.vertexsets, set_name) && throw(ArgumentError("Electrode vertexset name $(set_name) already exists"))
+    addvertexset!(mesh, set_name*"-p", x -> x ≈ positive_electrode)
+    addvertexset!(mesh, set_name*"-n", x -> x ≈ negative_electrode)
+    positive_vertex = first(getvertexset(mesh, set_name*"-p"))
+    negative_vertex = first(getvertexset(mesh, set_name*"-n"))
+
+    local_positive_dof = Ferrite.vertexdof_indices(Ferrite.getfieldinterpolation(dh.subdofhandlers[1], :Z))[positive_vertex[2]][1]::Int
+    global_positive_dof = celldofs(dh, positive_vertex[1])[local_positive_dof]::Int
+    f[global_positive_dof] = -1
+    local_negative_dof = Ferrite.vertexdof_indices(Ferrite.getfieldinterpolation(dh.subdofhandlers[1], :Z))[negative_vertex[2]][1]::Int
+    global_negative_dof = celldofs(dh, negative_vertex[1])[local_negative_dof]::Int
+    f[global_negative_dof] = 1
+end
+
+"""
+    TODORENAMEECGPoissonReconstructionCache(mesh, κ, qr_collection, ip_collection, zero_vertex::VertexIndex)
```

Comment:
```suggestion
    ECGPoissonReconstructionCache(mesh, κ, qr_collection, ip_collection, zero_vertex::VertexIndex)
```

I think this was first proposed by https://ieeexplore.ieee.org/document/4015619 . We should also mention https://ieeexplore.ieee.org/abstract/document/5759732 as a method to improve the precision further. The FIMH paper is another analysis trying to explain what exactly breaks down.